pages/_app.js

```diff
@@ -1,10 +1,36 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
+import { initialPlants } from "@/assets/plants";
 
 export default function App({ Component, pageProps }) {
+  const [plants, setPlants] = useState(initialPlants);
+  const [plantsInfo, setPlantsInfo] = useState(
+    plants.map((plant) => ({ ...plant, isMyPlant: false }))
+  );
```

Comment:
Duplicating the complete state is very inefficient and can lead to a lot of bugs. There are two possible better solutions: 
1. add 'isMyPlant: false' to your 'initialPlants' and you're fine.
2. keep only the additional data in the 'plantsInfo' state. This solution is a bit more flexible for the future e.g. if you want to save your favourites in localStorage